### PR TITLE
CVSB-12566

### PR DESCRIPTION
**Pull request by vladddragomir**

### CODE REVIEW

File: src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ export class MultipleTechRecordsSelectionPage {
     this.isAtLeastOneSkeleton = this.vehicles.some(vehicle => {
       return this.vehicleService.isVehicleSkeleton(vehicle);
     });
+    this.vehicleService.sortSkeletonVehicles(this.vehicles);
```
Comment (agoncearenco) [inline src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts]: Why not sorting it here https://github.com/dvsa/cvs-app-mobile/blob/6227009436461ed70bd8f986ee3a03670643547f/src/providers/vehicle/vehicle.service.ts#L81? At the moment half of the sorting is done in the service and the second half is done in this page

Comment (vladddragomir) [reply src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts]: ok, right! i haven't seen your sorting in the service

File: src/providers/vehicle/vehicle.service.ts [hunk 2]
```diff
@@ -142,4 +142,18 @@ export class VehicleService {
...
+      return -1;
+    }
+    if (this.isVehicleSkeleton(vehicle1) && !this.isVehicleSkeleton(vehicle2)) {
+      return 1;
+    }
+    return 0;
+  }
+
+  sortSkeletonVehicles(vehicles: VehicleModel[]) {
+    vehicles.sort((a, b) => this.compareSkeletonVehicles(a, b));
```
Comment (KHodderDVSA) [inline src/providers/vehicle/vehicle.service.ts]: should just be able to pass the function directly:
vehicles.sort(this.compareSkeletonVehicles);

File: test-config/services-mocks/vehicle-service.mock.ts [hunk 3]
```diff
@@ -100,4 +100,12 @@ export class VehicleServiceMock {
...
     return this.vehicleService.isVehicleSkeleton(vehicle);
   }
+
+  compareSkeletonVehicles(vehicle1: VehicleModel, vehicle2: VehicleModel): number {
+    return this.vehicleService.compareSkeletonVehicles(vehicle1, vehicle2);
+  }
+
+  sortSkeletonVehicles(vehicles: VehicleModel[]) {
+    vehicles.sort((a, b) => this.compareSkeletonVehicles(a, b));
+  }
```
Comment (KHodderDVSA) [inline test-config/services-mocks/vehicle-service.mock.ts]: no duplicate code. Use the originals. Did that for compareSkeletonVehicles, but you don't actually need it all - just use this.vehicleService.sortSkeletonRecords directly

File: src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts (hunk 1 above)
Comment (KHodderDVSA) [reply src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts]: Agree with this ^

File: test-config/services-mocks/vehicle-service.mock.ts (hunk 3 above)
Comment (vladddragomir) [reply test-config/services-mocks/vehicle-service.mock.ts]: refactored

File: src/providers/vehicle/vehicle.service.ts (hunk 2 above)
Comment (vladddragomir) [reply src/providers/vehicle/vehicle.service.ts]: refactored

File: src/providers/vehicle/vehicle.service.spec.ts [hunk 4]
```diff
@@ -185,4 +185,28 @@ describe('Provider: VehicleService', () => {
...
+    let vehicle2 = {...VehicleDataMock.VehicleData};
+    vehicle1.techRecord.recordCompleteness = 'skeleton';
+    vehicle2.techRecord.recordCompleteness = 'complete';
+    expect(vehicleService.compareVehicles(vehicle1, vehicle2)).toEqual(1);
+  });
+
+  it('should compare two techRecords and return -1 if the first vehicle should alphabetically be before the second one', () => {
+    let vehicle1 = {...VehicleDataMock.VehicleData};
+    let vehicle2 = {...VehicleDataMock.VehicleData};
+    vehicle1.techRecord.chassisMake = 'T';
```
Comment (KHodderDVSA) [inline src/providers/vehicle/vehicle.service.spec.ts]: Is this supposed to be recordCompleteness?

File: src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts (hunk 1 above)
Comment (vladddragomir) [reply src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts]: refactored after synched with Alex's logic

File: src/providers/vehicle/vehicle.service.spec.ts (hunk 4 above)
Comment (vladddragomir) [reply src/providers/vehicle/vehicle.service.spec.ts]: there are 3 tests. first 2 are about recordCompleteness

Comment (vladddragomir) [reply src/providers/vehicle/vehicle.service.spec.ts]: this is the comparison function that captures both my and Alex's logic

Comment (KHodderDVSA) [reply src/providers/vehicle/vehicle.service.spec.ts]: I understand.... but the next line has recordCompleteness = 'W' so, not sure what case this is addressing.

Comment (vladddragomir) [reply src/providers/vehicle/vehicle.service.spec.ts]: holly s**t .... right

Comment (vladddragomir) [reply src/providers/vehicle/vehicle.service.spec.ts]: fixed

**Pull request merged**